src/test/java/com/google/api/generator/engine/writer/ImportWriterVisitorTest.java

```diff
@@ -54,6 +55,71 @@ public void setUp() {
     writerVisitor = new ImportWriterVisitor(CURRENT_PACKAGE);
   }
 
+  @Test
+  public void writeNewObjectExprImports_genericsAndVariableArgs() {
+    // [Constructing] new HashMap<List<String>, IOException>>(int initialCapacity, float loadFactor)
```

Comment:
Consider a test where the argument types need to be imported. Feel free to either modify this test or make another one. 

Also, our sample code should strive to be valid Java if possible. For instance, passing an exception type to the constructor of this HashMap would not be valid Java, so we should avoid doing that in tests.

Reply:
sure, the test below `writeNewObjectExprImports_methodExprArg` covers the case that `new IOException(message, cause(mapArg))` where the second arg `cause(mapArg)` is a method which takes a `hashmap` and return a `throwable`, and `HashMap` is asserted to be imported for this case.

Sounds reasonable to me, changed the generics for HashMap to be `<List<String>, Integer>`